### PR TITLE
layout: use `Au` in replaced elements

### PR DESCRIPTION
**Pull request by atbrakhi**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

### CODE REVIEW

**Comment by atbrakhi** [comment]
not ready for review yet, I see more opportunity to use `au` in replaced elements but i am also expecting some wpt test failures, let see!

**Comment by Loirooriol** [comment]
If you comment out `rustc = "crown"` in `.cargo/config.toml`, then you can use `cargo clippy --fix -p layout_2020 --allow-dirty --broken-code` to automatically avoid unnecessary `into()`.

File: components/layout_2020/replaced.rs [hunk 1]
```diff
@@ -406,15 +409,15 @@ impl ReplacedContent {
...
                         // equation used for block-level, non-replaced elements in normal flow.”
-                        _ => default_object_size().inline,
+                        _ => default_object_size().inline.into(),
                     };
                 let block_size = if let Some(block) = intrinsic_size.block {
                     block
                 } else if let Some(i_over_b) = intrinsic_ratio {
                     // “used width” in CSS 2 is what we just computed above
-                    inline_size / i_over_b
+                    inline_size / i_over_b as i32
```
Comment (Loirooriol) [inline components/layout_2020/replaced.rs:418]: Seems better to use `scale_by(1.0 / i_over_b)`, e.g.

```rs
let inline_size = Au::from_px(1);
let i_over_b = 31.0;
println!("{:?}", inline_size / i_over_b as i32);        // 0.016666666666666666px
println!("{:?}", inline_size.scale_by(1.0 / i_over_b)); // 0.03333333333333333px
```

And in math `1 / 31` is closer to the latter.

File: components/layout_2020/replaced.rs [hunk 2]
```diff
@@ -387,7 +390,7 @@ impl ReplacedContent {
                         (None, Some(block), Some(i_over_b)) => {
                             // “used height” in CSS 2 is always gonna be the intrinsic one,
                             // since it is available.
-                            block * i_over_b
+                            block * i_over_b as i32
```
Comment (Loirooriol) [inline components/layout_2020/replaced.rs:393]: Ditto, use `scale_by()`

**Review by Loirooriol — CHANGES_REQUESTED** [review]

File: components/layout_2020/replaced.rs (hunk 1 above)
Comment (atbrakhi) [reply components/layout_2020/replaced.rs]: Thanks for the tip. `i_over_b as i32` did made me uncomfortable. Updated in [28cfcbe](https://github.com/servo/servo/pull/31178/commits/28cfcbe7048a152c952043bf73054126c2b38169)

File: components/layout_2020/replaced.rs [hunk 3]
```diff
@@ -387,7 +390,7 @@ impl ReplacedContent {
                         (None, Some(block), Some(i_over_b)) => {
                             // “used height” in CSS 2 is always gonna be the intrinsic one,
                             // since it is available.
-                            block * i_over_b
+                            block.scale_by(1.0 * i_over_b)
```
Comment (Loirooriol) [inline components/layout_2020/replaced.rs:393]:
```suggestion
                            block.scale_by(i_over_b)
```

File: components/layout_2020/replaced.rs [hunk 4]
```diff
@@ -431,15 +434,15 @@ impl ReplacedContent {
                     Below(Length),
                     Above(Length),
                 }
-                let violation = |size, min_size, mut max_size: Option<Length>| {
+                let violation = |size: Au, min_size, mut max_size: Option<Length>| {
```
Comment (Loirooriol) [inline components/layout_2020/replaced.rs:437]: Mixing Au and Length here is a bit weird but I guess it will improve when more things become Au

**Review by Loirooriol — APPROVED** [review]

**Comment by sagudev** [comment]
> If you comment out `rustc = "crown"` in `.cargo/config.toml`, then you can use `cargo clippy --fix -p layout_2020 --allow-dirty --broken-code` to automatically avoid unnecessary `into()`.

This is probably needed due to clippy being rustc driver too (like crown).

**Pull request merged**